src/sphmultipcqueue.cpp

```diff
@@ -325,6 +325,7 @@ SPHMPMCQAdvanceHead(SPHLFEntryHeader_t **head_p, sphLFEntryID_t idAlloc, sphLFEn
 	return 0;
 }
 
+#if defined(HAS_GNU_TM) || defined(__HTM__)
```

Comment:
What's the preference on how to do that?
I suppose currently, the test will fail to build, which is a clear indication to the user that the MPMC implementation is not available for this build.  So, there is some value in keeping it as it is.
Allow the test to run, but respect the defines in the test case, so the test doesn't really do anything with the API.  Actually, the MPSC and SPMC APIs are still valid, so that test case could run with that subset of tests.  Specifically, the tests that essentially are SPSC tests (using the SPMC on the producer side and MPSC on the consumer side.)
I'll submit those changes for your consideration.